### ISSUE TITLE
sigma_const hyper parameter 

### ISSUE THREAD

**Issue by motokimura** [jwchoi384/Gaussian_YOLOv3#40]
Hi, I talked with you at your poster in ICCV2019!
I just released [PyTorch implementation of Gaussian YOLOv3](https://github.com/motokimura/PyTorch_Gaussian_YOLOv3) with training code on COCO dataset. Would it be possible to include link to our repo in [third-party-implementations section](https://github.com/jwchoi384/Gaussian_YOLOv3#third-party-implementations) of your README? If it is okay for you, I can send a pull request to update your README.

Though Gaussian YOLOv3 in our repo shows significant improvement of COCO mAP (2.7 point) on COCO2017 val, this improvement is still smaller than the one reported in your paper (3.1 point).
I'm wondering if this difference comes from the hyper parameter [`sigma_const`](https://github.com/jwchoi384/Gaussian_YOLOv3/blob/7d2ab7c927414f912a647c93ce807fbfb448255e/src/gaussian_yolo_layer.c#L104) set to 0.3 in your implementation (our implementation does not have this parameter).

Do you think `sigma_const` affects the result a lot?
How did you find the value 0.3 for this parameter?

**Comment by jwchoi384** [comment]
@motokimura 
Hi, I remember you! Nice to meet you!
Thanks for your implementation 👍 
Please send me pull request for update the README.

I experimented with several `sigma_const` values, and there was some difference.
On BDD validation set (over 200k iteration weight),
sigma_const: 0.1 -> mAP is 18.49-18.65.
sigma_const: 0.2 -> mAP is 18.68-19.11.
sigma_const: 0.3 -> mAP is 18.92-19.14 (and when we reduce the learning rate in on-going training, we can get over 19.7 mAP)
Of those, the value of 0.3 was best.
If sigma is not added to the loss in our C implementation, the training did not work well. 
I don't know exactly why, but it seems to be sensitive to variance.
So i added it my model.

**Comment by motokimura** [comment]
Glad to see you there, too! 😄
Thank you so much for merging PR!

`sigma_const` looks more important than I thought.
I agree loss is sensitive to variance with small `sigma_const`. In our experiments without `sigma_const` on COCO dataset, we needed gradient clipping to avoid divergence. Also, mAP increased more slowly than when we trained normal YOLOv3 with the same hyper parameters [[link]](https://github.com/motokimura/PyTorch_Gaussian_YOLOv3#performance). These may be caused by large magnitude of gradients and `sigma_const` seems to mitigate this kind of instability of the gaussian loss. 

I will try Gaussian YOLOv3 training on COCO again with `sigma_const` parameter for higher mAP!

**Comment by jwchoi384** [comment]
@motokimura 
I see! Thanks!

**Comment by motokimura** [comment]
I'll let you konw when I got the experiment result! Thanks for your kind answers!

**Comment by CuongNguyen218** [comment]
@motokimura , can you open issues in your repo ?

**Comment by NewRGB** [comment]
Hi Jiwoong,
                It is received an e-mail from the
Github/jwchoi384/Gaussian_YOLOv3.
I'm in Github and forked your repository (
https://github.com/jwchoi384/Gaussian_YOLOv3), yesterday.
Your project is very very interesting and very new concept. I'm very much
interested to learn about this project about autonomous vehicle.
Could you please, guide/mentor me on this project ?

Your response is appreciated.
-------------------------------
*Thanks & Regards,*
*-Amber*
=============================================================

**Comment by motokimura** [comment]
@CuongNguyen218 
I enabled `issues` feature in our repo! 
https://github.com/motokimura/PyTorch_Gaussian_YOLOv3/issues

I forgot to enable this feature.
Thanks for your comment!
